OpenRA.Mods.Common/Widgets/Logic/ColorPickerLogic.cs

```diff
@@ -25,11 +25,13 @@ public class ColorPickerLogic : ChromeLogic
 		int paletteTabHighlighted = 0;
 
 		[ObjectCreator.UseCtor]
-		public ColorPickerLogic(Widget widget, ModData modData, World world, HSLColor initialColor, Action<HSLColor> onChange, Dictionary<string, MiniYaml> logicArgs)
+		public ColorPickerLogic(Widget widget, ModData modData, World world, HSLColor initialColor, string initialFaction, Action<HSLColor> onChange,
+			Dictionary<string, MiniYaml> logicArgs)
 		{
 			string actorType;
-			if (!ChromeMetrics.TryGet("ColorPickerActorType", out actorType))
-				actorType = "mcv";
+			if (initialFaction == null || !ChromeMetrics.TryGet("ColorPickerActorType-" + initialFaction, out actorType))
+				if (!ChromeMetrics.TryGet("ColorPickerActorType", out actorType))
+					actorType = "mcv";
```

Comment:
IMO this would be a good time to remove this mod-specific default, and require all mods to list it explicitly.

Reply:
removing "mcv" makes sense, but the default "ColorPickerActorType" is required for random faction or the color picker in the settings.

Reply:
Yup, indeed.